sklearn/ensemble/_bagging.py

```diff
@@ -393,10 +383,6 @@ def fit(self, X, y, *, sample_weight=None, **fit_params):
             multi_output=True,
         )
 
-        if sample_weight is not None:
-            sample_weight = _check_sample_weight(sample_weight, X, dtype=None)
-            fit_params["sample_weight"] = sample_weight
-
```

Comment:
The `allow` suggestion didn't work here but I removed this validation because `bagging` doesn't consume sample_weight. So it should delegate validation to the underlying estimator.

Otherwise, if it turns out that bagging should be a consumer, we'll need to put back sample_weight as explicit param, right ?

Reply:
+1 to keep sample_weight as explicit param for this release, as indeed we will need them for a PR fixing sample weight in Bagging estimators.

Reply:
I don't understand how that affects your work, can you elaborate?

Reply:
It's for https://github.com/scikit-learn/scikit-learn/pull/31165 where we use `sample_weight` for drawing the samples instead of passing them to the subestimators. Bagging is then is a consumer of `sample_weight`, and consumer-only as it should not pass them to underlying estimators.

Reply:
Oh interesting. I'd say we can merge this one, and your PR can add it as an explicit parameter. Nothing changes for the user's code, since they can pass `sample_weight=sample_weight` regardless of it being a part of `kwargs` or an explicit parameter.